eval/ph/perceval_simulation.py

```
@@ -0,0 +1,240 @@
+from perceval.components import BS, PS
+
+if TYPE_CHECKING:
+    import torch
+
+
+def create_mzi_chip(
+    bs_list: np.ndarray,
+    ps_matrix: torch.Tensor | np.ndarray,
+    phase_error: float | None,
+    chip_size: int,
+    exclude_edge_phase_shifters: bool = False,
```

Ruff (0.15.21)

[warning] Boolean-typed positional argument in function definition

(FBT001)